src/mlpack/tests/lsh_test.cpp

```diff
@@ -9,103 +9,211 @@
 #include "old_boost_test_definitions.hpp"
 
 #include <mlpack/methods/lsh/lsh_search.hpp>
+#include <mlpack/methods/neighbor_search/neighbor_search.hpp>
 
 using namespace std;
 using namespace mlpack;
 using namespace mlpack::neighbor;
 
+double compute_recall(
+    arma::Mat<size_t> LSHneighbors, 
+    arma::Mat<size_t> groundTruth)
+{
+  const int n_queries = LSHneighbors.n_cols;
+  const int n_neigh = LSHneighbors.n_rows;
+
+  int found_same = 0;
+  for (int q = 0; q < n_queries; ++q)
+  {
+    for (int n = 0; n < n_neigh; ++n)
```

Comment:
Another minor note: maybe better to use `size_t` here since these will never take negative values?